File: Algo_Method/multiple_algo/Q2-2/Q2-2.py
def f(n, x):
    n += 1
    for i in range(5): n = n * x + 1
    return n

n, m = map(int, input().split())
l, r = -1, 101
while r - l > 0.0001:
    mid = (l + r) / 2.0
    if f(n, mid) > m: r = mid
    else: l = mid
print(l)

File: Algo_Method/multiple_algo/Q2-2/Q2-2.rb
def f(n, x)
    n += 1
    5.times do
        n = n * x + 1 
    end
    return n
end

n, m = gets.chomp.split.map(&:to_i)

l = -1
r = 101
while r - l > 0.0001 do
    mid = (l + r) / 2.0
    if f(n, mid) > m
        r = mid
    else
        l = mid
    end
end
p l